airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3ConsumerFactory.kt

```diff
@@ -117,7 +122,8 @@ class S3ConsumerFactory {
                         storageOperations.uploadRecordsToBucket(
                             writer,
                             writeConfig.namespace,
-                            writeConfig.fullOutputPath
+                            writeConfig.fullOutputPath,
+                            writeConfig.generationId
```

Comment:
Later up in the stack, I'll be failing fast if these are null similar to what we do in `CatalogParser`.